File: backend/src/main/java/com/example/stamp_app/entity/MicroController.java
package com.example.stamp_app.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Pattern;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.ColumnDefault;
import org.hibernate.annotations.Comment;

import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

@Entity
@Getter
@AllArgsConstructor
@NoArgsConstructor
public class MicroController {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotNull
    @Column
    @Comment(value = "端末UUID")
    private UUID uuid;

    @Column
    @Comment(value = "端末名")
    private String name;

    @NotNull
    @Comment(value = "MACアドレス")
    @Pattern(regexp = "^([0-9A-Fa-f]{2}[:-]){5}[0-9A-Fa-f]{2}$")
    private String macAddress;

    @Column
    @ColumnDefault("60")
    @Comment(value = "測定間隔(分)")
    @Pattern(regexp = "^(60|30|20|15|10|5|1)$")
    private String interval;

    @Column
    @Comment(value = "測定に使用するSDI-12アドレス(カンマ区切り)")
    @Pattern(regexp = "^((([0-9A-Za-z]{1},)+[0-9A-za-z]{1})|([0-9A-za-z]{1})?)$")
    private String sdi12Address;

    @Column
    private LocalDateTime createdAt;

    @Column
    private LocalDateTime updatedAt;

    @Column
    private LocalDateTime deletedAt;

    @OneToMany(mappedBy = "microController")
    private List<MeasuredDataMaster> measuredDataMasters;

    @ManyToOne
    private Account account;

}

File: backend/src/main/java/com/example/stamp_app/entity/MicroController.kt
package com.example.stamp_app.entity

import jakarta.persistence.*
import jakarta.validation.constraints.NotNull
import jakarta.validation.constraints.Pattern
import org.hibernate.annotations.ColumnDefault
import org.hibernate.annotations.Comment
import java.time.LocalDateTime
import java.util.*

@Entity
data class MicroController(
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	val id: Long? = null,

	@Column
	@Comment(value = "端末UUID")
	val uuid: @NotNull UUID? = null,

	@Column
	@Comment(value = "端末名")
	val name: String? = null,

	@Comment(value = "MACアドレス")
	val macAddress: @NotNull @Pattern(regexp = "^([0-9A-Fa-f]{2}[:-]){5}[0-9A-Fa-f]{2}$") String? = null,

	@Column
	@ColumnDefault("60")
	@Comment(value = "測定間隔(分)")
	val interval: @Pattern(regexp = "^(60|30|20|15|10|5|1)$") String,

	@Column
	@Comment(value = "測定に使用するSDI-12アドレス(カンマ区切り)")
	val sdi12Address: @Pattern(regexp = "^((([0-9A-Za-z]{1},)+[0-9A-za-z]{1})|([0-9A-za-z]{1})?)$") String? = null,

	@Column
	val createdAt: LocalDateTime? = null,

	@Column
	val updatedAt: LocalDateTime? = null,

	@Column
	val deletedAt: LocalDateTime? = null,

	@OneToMany(mappedBy = "microController")
	val measuredDataMasters: List<MeasuredDataMaster>? = null,

	@ManyToOne
	val account: Account? = null
)
